# HG changeset patch
# User andrew
# Date 1502409945 -3600
# Node ID 64e09337b9b7d253243e6160660e025b7269e04d
# Parent  0c14af84f624bce3b6eafa8dedc1b4b4e85ab1f9
8186090: java.nio.Bits.unaligned() doesn't handle aarch64
Summary: Check for "aarch64" along with other unaligned access supporting architectures.
Reviewed-by: andrew
Contributed-by: Derek White <Derek.White@cavium.com>

--- a/jdk/src/share/classes/java/nio/Bits.java
+++ b/jdk/src/share/classes/java/nio/Bits.java
@@ -615,7 +615,8 @@ class Bits {
             new sun.security.action.GetPropertyAction("os.arch"));
         unaligned = arch.equals("i386") || arch.equals("x86")
             || arch.equals("amd64") || arch.equals("x86_64")
-            || arch.equals("ppc64") || arch.equals("ppc64le");
+            || arch.equals("ppc64") || arch.equals("ppc64le")
+            || arch.equals("aarch64");
         unalignedKnown = true;
         return unaligned;
     }
--- a/jdk/src/share/classes/sun/security/provider/ByteArrayAccess.java
+++ b/jdk/src/share/classes/sun/security/provider/ByteArrayAccess.java
@@ -94,7 +94,8 @@ final class ByteArrayAccess {
         String arch = java.security.AccessController.doPrivileged
             (new sun.security.action.GetPropertyAction("os.arch", ""));
         return arch.equals("i386") || arch.equals("x86") || arch.equals("amd64")
-            || arch.equals("x86_64") || arch.equals("ppc64") || arch.equals("ppc64le");
+            || arch.equals("x86_64") || arch.equals("ppc64") || arch.equals("ppc64le")
+            || arch.equals("aarch64");
     }
 
     /**
